llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -7362,6 +7363,18 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
   else if (auto *ES = dyn_cast<ExternalSymbolSDNode>(CLI.Callee))
     CalleeAttrs = SMEAttrs(ES->getSymbol());
 
+  auto DescribeCallsite =
+      [&](OptimizationRemarkAnalysis &R) -> OptimizationRemarkAnalysis & {
```

Comment:
nit: Would it be easier to have a function that takes just a StringRef, so that you can do `EmitSMERemarkForCall("requires a streaming mode transition");`? You can then hide the construction of the ORE and OptimizationRemarkAnalysis in this lambda.

Reply:
I was thinking the current format would be a bit nicer for additions, i.e. if/when ZA liveness starts being tracked, the remark could be amended to mention the number of za slices that should be preserved.

Reply:
I'm a bit weary of writing code in anticipation of future changes which may never happen. We can always rewrite the code in this form later if that makes more sense. That said, I don't feel too strongly about it since the current code isn't bad either.